kb/deploy-new-version-of-workflow.md

```diff
@@ -0,0 +1,70 @@
+---
+slug: deploy-new-version-of-workflow
+title: Use Replay before deploy a new version of a Workflow
+tags:
+  - troubleshooting
+  - how-to
+date: 2023-05-11T00:00:00Z
+---
+
+When making code changes to a Workflow, it is important to avoid introducing non-deterministic behavior.
+
+One way to prevent this error from occurring is to use the [Workflow Replayer](/workflows#replays) to replay existing Workflow Histories against your new code before deploying it to production.
+
+:::important
+
+While the Workflow Replay helps to avoid non-deterministic code, it does not guarantee that your code is deterministic.
+
+:::
+
+## How to use the Workflow Replayer
+
+To use the Workflow Replayer, you need to provide the following:
+
+- The Workflow Replayer to register the Workflow that you want to replay
+- A Workflow History to execute, either:
+  - Loaded directly from the Client
+  - Loaded from the JSON History file
+
+Consult the [SDK specific documentation](/application-development/testing#replay) for information on using the Workflow Replayer programmatically.
+
+If you chose to load the Event History from a JSON file on disk, you can do so programmatically or from one of the following methods:
+
+### The Temporal CLI
+
+Use the Temporal CLI to download the Workflow History file from the CLI:
+
+```command
+temporal workflow show --workflow-id <workflow_id> --output json > <output_file>
+```
+
+### The Temporal Web UI
+
+Navigate through the Temporal Web UI to download the Workflow History file:
+
+1. Navigate to the Workflow Execution you want to download the History from.
+2. In the **Recent Events** section, select **Download**.
+
+Once you have the Workflow JSON History file, you can use the Workflow Replayer to replay the Workflow History, in the SDK of your choice.
+
+## Replay existing Workflow Histories in CI
```

Comment:
This is core documentation that should live here: https://docs.temporal.io/workflows#replays

This explains a significant use case for the Replayer which is a key aspect of explaining it as a concept -